libraries/lib-registries/Registry.cpp

```diff
@@ -603,9 +603,12 @@ auto CollectedItems::MergeItems(
                force = true;
          }
       }
-      else if ( iPass == OrderingHint::End && endItemsCount == 0 )
+      else if (iPass == OrderingHint::End && endItemsCount == 0)
+      {
+         assert(newSize >= prevSize || newSize == 0);
```

Comment:
Thanks for this assertion which explains that crash you saw.  I wanted to know how to reproduce this.  So, it was in resetting the configuration.

I reproduced the assertion failure at the commit before this one. Then  I rebased my long waiting registry-changes draft PR branch onto it and tried again, and confirmed that the problem is fixed.

Sorry for the pain that the withholding of that fixed caused.  Let's get that branch self-reviewed, and reviewed, and merge early in 3.4 so the Registry system is more reliable.